motr/io_nw_xfer.c

```diff
@@ -873,9 +879,32 @@ static int target_ioreq_iofops_prepare(struct target_ioreq *ti,
 				else
 					buf = bvec->ov_buf[seg];
 
-				rc = m0_rpc_bulk_buf_databuf_add(rbuf,
-					buf, COUNT(ivec, seg),
-					INDEX(ivec, seg), ndom);
+				xfer_len = COUNT(ivec, seg);
+				offset = INDEX(ivec, seg);
+				
+				/*
+				 * Accomodate multiple pages in a single 
```

Comment:
s/Accomodate/Accommodate

Reply:
Please add comments here to explain why we need to do this, specifically for libfabric.

If multiple pages are squeezed into a single net buffer, do we need to squeeze the bulk descriptor in rpc message?

Reply:
> s/Accomodate/Accommodate

Corrected



> Please add comments here to explain why we need to do this, specifically for libfabric.
> 
> If multiple pages are squeezed into a single net buffer, do we need to squeeze the bulk descriptor in rpc message?

This is not specifically for libfabric transport. This can be generic for all the transports which can have segment size > page size. (If we set the segment size in libfabric to 4k, the above code would have no impact)
Multiple contiguous pages can be treated as a single contiguous chunk of memory (as long as it fits in one segment) and this segment can be transferred by the transport in a single go.
Hence we would be generating only one descriptor per segment instead of one descriptor per page.